src/controllers/share/share.controller.ts

```
@@ -0,0 +1,95 @@
+    // Validate the header hash matches claimed difficulty
+    const headerBuffer = Buffer.from(submission.header, 'hex');
+    const hashResult = bitcoinjs.crypto.hash256(headerBuffer);
+    const difficulty = this.calculateDifficulty(hashResult);
+
+    // Verify the calculated difficulty matches or exceeds claimed difficulty
+    if (difficulty < this.minimumDifficulty) {
+      throw new UnauthorizedException('Share difficulty too low');
+    }
+
+    const timestamp = headerBuffer.readUInt32LE(68);
+    const tenMinutesAgo = Math.floor(Date.now() / 1000) - (10 * 60);
```

ESLint

[error] Replace `(10·*·60)` with `10·*·60`

(prettier/prettier)